pkg/operator/ceohelpers/util.go

```diff
@@ -0,0 +1,54 @@
+package ceohelpers
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+
+	"github.com/ghodss/yaml"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/klog/v2"
+)
+
+const (
+	clusterConfigName      = "cluster-config-v1"
+	clusterConfigKey       = "install-config"
+	clusterConfigNamespace = "kube-system"
+)
+
+type replicaCountDecoder struct {
+	ControlPlane struct {
+		Replicas string `yaml:"replicas,omitempty"`
+	} `yaml:"controlPlane,omitempty"`
+}
+
+// GetMastersReplicaCount get number of expected masters statically defined by the controlPlane replicas in the install-config.
+func GetMastersReplicaCount(ctx context.Context, kubeClient kubernetes.Interface) (int, error) {
+	klog.Infof("Getting number of expected masters from %s", clusterConfigName)
+	clusterConfig, err := kubeClient.CoreV1().ConfigMaps(clusterConfigNamespace).Get(ctx, clusterConfigName, metav1.GetOptions{})
+	if err != nil {
+		klog.Errorf("Failed to get ConfigMap %s: %v", clusterConfigName, err)
```

Comment:
don't log errors and return them at the same time. That's an anti-pattern. Same below.